File: backend/app/schemas/invitation.py
# app/schemas/invitation.py
from pydantic import BaseModel, EmailStr, Field
from typing import Optional
from datetime import datetime
# Import the enum from models to be used in Pydantic schema
from app.models.invitation import InvitationStatusEnum 

class InvitationBase(BaseModel):
    email: EmailStr
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    role_to_assign: str = Field("staff", description="Role to assign to the invited user, defaults to 'staff'")

class InvitationCreate(InvitationBase):
    # tenant_id will be derived from the inviting admin's context
    pass

class InvitationAccept(BaseModel):
    token: str
    password: str = Field(min_length=8) # Add password validation
    # first_name and last_name might be optional if already provided in invite,
    # but good to allow override/confirmation here.
    first_name: str = Field(min_length=1) 
    last_name: str = Field(min_length=1)

class InvitationOut(InvitationBase):
    id: int
    tenant_id: int
    status: InvitationStatusEnum # Use the enum here
    token_expiry: datetime
    invited_by_user_id: int
    accepted_by_user_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
        use_enum_values = True # Ensures enum values (strings) are used in serialization

class ValidateTokenResponseSchema(BaseModel):
    valid: bool
    # These fields are populated if valid = True
    email: Optional[EmailStr] = None
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    role: Optional[str] = None # Role assigned in the invitation (e.g., "staff", "admin")
    
    # These fields are populated if valid = False
    error_code: Optional[str] = None # E.g., "EXPIRED", "NOT_FOUND", "ALREADY_ACCEPTED", "CANCELLED", "INVALID_STATUS"
    message: Optional[str] = None    # Human-readable error message to display on the frontend

File: frontend/src/types/Invitation.ts
// src/types/Invitation.ts
// This file contains TypeScript interfaces and types related to invitations.

export type InvitationStatus = 'pending' | 'accepted' | 'expired' | 'cancelled';

export interface InvitationBase {
    email: string;
    first_name?: string | null;
    last_name?: string | null;
    role_to_assign: string; // e.g., "staff", "admin"
}

export interface InvitationCreatePayload extends InvitationBase {
    // tenant_id is handled by backend context
}


export interface InvitationAcceptPayload {
    token: string;
    first_name: string;
    last_name: string;
    password: string;
}

export interface InvitationOut extends InvitationBase {
    id: number;
    tenant_id: number;
    status: InvitationStatus;
    token_expiry: string; // ISO date string
    invited_by_user_id: number;
    accepted_by_user_id?: number | null;
    created_at: string; // ISO date string
    updated_at: string; // ISO date string
}

// For the GET /staff/invitations/validate-token endpoint
export interface ValidateTokenResponse {
    valid: boolean;
    email?: string;
    first_name?: string | null;
    last_name?: string | null;
    role?: string; // Role assigned in the invitation
    error_code?: 'EXPIRED' | 'NOT_FOUND' | 'ALREADY_ACCEPTED' | 'CANCELLED' | 'INVALID_STATUS';
    message?: string;
}
